### PR TITLE
[stable/gocd] add annotations for deployment

### PR DESCRIPTION
**Pull request by skloss**
#### What this PR does / why we need it:
This annotations loop for the k8s deployment object to use GOCD  in k8s with annotations like kubeaim.
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:
@arvindsv 
@bdpiparva
@GaneshSPatil
@varshavaradarajan
@maheshp
@sheroy
#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/gocd/templates/gocd-agent-deployment.yaml [hunk 1]
```diff
@@ -8,6 +8,12 @@ metadata:
     release: {{ .Release.Name | quote }}
     heritage: {{ .Release.Service | quote }}
     component: agent
+    {{- if .Values.server.deployment.annotations }}
```
Comment (varshavaradarajan) [inline stable/gocd/templates/gocd-agent-deployment.yaml]: This should be `.Values.agent.deployment.annotations`. @davidkarlsen , @skloss

**Comment by arvindsv** [comment]
Trying to cancel it, since there might be a bug, as @varshavaradarajan said.

**Comment by varshavaradarajan** [comment]
@davidkarlsen - please update the changelog too. Thanks!

**Pull request closed without merging**